src/Host/Client/Impl/Session/RSession.cs

```diff
@@ -551,7 +554,12 @@ public Task StartHostAsync(RHostStartupInfo startupInfo, IRSessionCallback callb
 
             var callback = _callback;
             if (!addToHistory && callback != null) {
-                return await callback.ReadUserInput(prompt, len, ct);
+                try {
+                    IsReadingUserInput = true;
+                    return await callback.ReadUserInput(prompt, len, ct);
```

Comment:
Yes, if you do a reentrable eval while on `ReadConsole`, and that eval involves some code that asks for input (e.g. another `readline`).